backend/matching/src/services/get-match-items.ts

```diff
@@ -1,7 +1,133 @@
-export const getMatchItems = () => {
-  // TODO: Get Match Items
-  return {
-    roomId: '',
-    questionId: '',
-  };
+import { IMatchType } from "@/types";
+import axios from 'axios';
+
+interface IGetRandomQuestionPayload {
+  attemptedQuestions: number[];
+  difficulty?: string;
+  topic?: string;
+}
+
+
+interface IServiceResponse<T> {
+    success: boolean;
+    data?: T;
+    error?: { message: string };
+  }
+
+interface IQuestion {
+  id: number;
+  title: string;
+  description: string;
+  difficulty: string;
+  topic: string[];
+}
+
+interface IMatchItemsResponse {
+    roomName: string;
+    questionId: number;
+    question: IQuestion;
+}
+
+export const getMatchItems = async (
+    searchIdentifier: IMatchType,
+    topic?: string,
+    difficulty?: string,
+    userId1?: string,
+    userId2?: string
+): Promise<IMatchItemsResponse | undefined> => {
+  const userEndpoint = `${process.env.USER_SERVER_ENDPOINT}`;
+  const questionEndpoint = `${process.env.QUESTION_SERVER_ENDPOINT}`;
+  const collabServerEndpoint = `${process.env.COLLAB_SERVER_ENDPOINT}`;
+
+  try {
+    if (!userId1 || !userId2) {
+      throw new Error('Both user IDs are required');
+    }
+
+    // Fetch attempted questions for both users
+    const [attemptedQuestions1, attemptedQuestions2] = await Promise.all([
+      fetchAttemptedQuestions(userEndpoint, userId1),
+      fetchAttemptedQuestions(userEndpoint, userId2)
```

Comment:
Is it better to fetch a single list given a list of userIds? Saves RTT of 2 requests and lets database handle it on user service end

Reply:
ehh, yes. did it this way cos it allows us to use the same endpoint on frontend, to check which questions have been attempted, which we don't have a route for yet, so might as well handle 2 cases at the same time. shdnt be a big deal to leave it as 2 queries.